features/0685-pickup-v2/README.md

```diff
@@ -0,0 +1,184 @@
+    "@type": "https://didcomm.org/messagepickup/2.0/status-request",
+    "recipient_key": "<key for messages>"
+}
+```
+
+- `recipient_key` is optional. When specified only return status related to that recipient key. This allows the _Recipient_ to discover if any messages are in the queue that were sent to a specific key.
+
+### Status
+
+Status details about pending messages
+
+```json=
+{
+    "@id": "123456781",
+    "@type": "https://didcomm.org/messagepickup/2.0/status",
+    "recipient_key": "<key for messages>",
+    "message_count": 7,
+    "duration_waited": 3600,
+    "newest_time": "2019-05-01 12:00:00Z",
+    "oldest_time": "2019-05-01 12:00:01Z",
+    "total_size": 8096,
+    "live_delivery": false
+}
+```
+
+`message_count` is the only required attribute. The others may be present if offered by the _message_holder_.
+
+`duration_waited` is in seconds, and is the longest  delay of any message in the queue.
+
+`total_size` is in bytes.
+
+If a `recipient_key` was specified in the status-request message, the matching value MUST be specified in the `recipient_key` attribute of the status message.
+
+`live_delivery` state is also indicated in the status message. 
+
+### Delivery Request
+
+A request from the _Recipient_ to the _Mediator_ to have waiting messages delivered. 
+Examples:
+
+```json=
+{
+    "@id": "123456781",
+    "@type": "https://didcomm.org/messagepickup/2.0/delivery-request",
+    "limit": 10,
+    "recipient_key": "<key for messages>"
+}
+```
+
+```json=
+{
+    "@type": "https://didcomm.org/messagepickup/2.0/delivery-request",
+    "limit": 1
+}
+```
+
+
+After receipt of this message, the mediator should deliver up to the limit indicated. 
+
+`recipient_key` is optional. When specified only return messages sent to that recipient key.
+
+If no messages are available to be sent, a status message MUST be sent immediately.
+
+Delivered messages MUST not be deleted until delivery is acknowledged by a `messages-received` message.
+
+
+### Message Received
+After receiving messages, the Recipient sends an ack message like this:
+```json=
+{
+    "@type": "https://didcomm.org/messagepickup/2.0/messages-received",
+    "message_tag_list": ["123","456"]
+}
+```
+
+`message_tag_list` identifiers are tags of each message received, extracted from the encrypted form of each message.
+
+Upon receipt of this message, the `Mediator` knows that the messages have been received, and can remove them from the collection of queued messages with confidence. The mediator should send an updated `status` message reflecting the changes to the queue.
+
+## Live Mode
+Live mode is the practice of delivering newly arriving messages directly to a connected _Recipient_. It is disabled by default and only activated by the _Recipient_. Messages that arrive when Live Mode is off MUST be stored in the queue for retrieval as described above.
+
+Live mode must only be enabled when a persistent transport is used, such as websockets.
```

Comment:
Can we say (I think this is implied below, but may need explicit mention) that if we're in live mode and the connection is broken, we automatically drop back to queued mode?

Reply:
I'd agree with explicitly mentioning it so that it doesn't get overlooked